proto/viam/app/data/v1/data.proto

```diff
@@ -80,6 +79,24 @@ message Filter {
   repeated string org_ids = 11;
   repeated string mime_type = 12;
   CaptureInterval interval = 13;
+  TagsFilter tags_filter = 14;
+  repeated string tags = 5 [deprecated = true];
+}
+
+message TagsFilter {
+  TagsFilterType type = 1;
+  repeated string tags = 2; // tags are used to match documents (using logical OR) if the TagsFilterType is TAGS_FILTER_TYPE_MATCH_BY_OR
+}
+
+// TagsFilterType specifies how data can be filtered based on tags
+enum TagsFilterType {
+  TAGS_FILTER_TYPE_UNSPECIFIED = 0;
+  // TAGS_FILTER_TYPE_MATCH_BY_OR specifies documents matched (using logical OR) on the tags field in the TagsFilter should be returned
```

Comment:
nit: think can remove "should be returned here" on this line to keep more succinct/readable